src/lib/stores/player.ts

```diff
@@ -46,18 +51,30 @@ const createInit = (): PlayerStore => {
 		crossFade,
 		dayPlayer,
 		nightPlayer,
+		progress: 0,
+		currentSongDuration: 100,
 		queue: {}
 	};
 };
 
 export function createPlayerStore() {
 	const init: PlayerStore | undefined = browser
 		? createInit()
-		: { fading: false, paused: false, queue: {} };
+		: { currentSongDuration: 100, fading: false, paused: false, progress: 0, queue: {} };
 
 	const { subscribe, update, set } = writable<PlayerStore>(init);
 
 	const updateAction = (newState: Partial<PlayerStore>) => {
+		if (newState.paused != undefined) {
+			handlePaused(newState.paused);
+		}
+		if (newState.currentPhaseSong != undefined) {
+			clearInterval(progressTimer);
```

Comment:
i don't really understand why you clear it and then directly start it again. We could just rename the progressTime to clock and keep it running all the time